### ISSUE TITLE
Remove AXI-Clock-Crossing module in top-level F1 file 

### ISSUE THREAD

**Issue by drichmond** [bespoke-silicon-group/bsg_replicant#510]
We currently have an AXI clock-crossing module in cosimulation so that we can switch IO/manycore clock frequencies. This has the effect of decreasing simulation time. 

However, it begs the question - do we really need it any more? After our new magic backdoors are added, is it still relevant? Does @leonardxiang's new module make it irrelevant?

So there are two options here: Remove it, because it's no longer needed. Replace it with something in BaseJump because it's blocking verilator.

I don't think it's the last verilator-blocking module. It's close though. I'm hopeful we no longer need it.

**Comment by mrutt92** [comment]
It might be still helpful when doing IO to other memories besides DRAM (like i-cache).

**Comment by drichmond** [comment]
Yeah. I'm hopeful Leonard's new module fixes it though. @leonardxiang Can you do some experiments with the clock switching enabled/disabled to find out how much your module helps?

**Comment by leonardxiang** [comment]
Take test_manycore_eva_read_write as an example:

**With new IO:**
clock switching   enable     /   disbled
The CPU time:     21.6s       /   23.7s
simulation time:  9.4 ms     /   6.4 ms
Note, if we optimize the clock conversion ratio for the new IO, more speedup will be achieved compared to the above results. (~17s, but not much)

**With Xilinx IP IO:**
The CPU time:     114.5s 
simulation time:   25.9 ms

In conclusion, the new IO is a lot faster than before (about X4 improvement).  And the benefit of clock switching is not that significant.






.

**Comment by drichmond** [comment]
Cool! Can you run all the regression tests and report the difference?

VCS will give you a runtime result for each test. You can use the following command: 

`find -name "test_*.log" | xargs -I{} grep -He "CPU Time" {} | sort`

**Comment by leonardxiang** [comment]
new_io_with_clock_switch
```
./cuda/test_barrier.log:CPU Time:     30.070 seconds;       Data structure size:   5.4Mb
./cuda/test_binary_load_buffer.log:CPU Time:     28.020 seconds;       Data structure size:   5.4Mb
./cuda/test_conv1d.log:CPU Time:     35.640 seconds;       Data structure size:   5.4Mb
./cuda/test_conv2d.log:CPU Time:     34.860 seconds;       Data structure size:   5.4Mb
./cuda/test_device_memcpy.log:CPU Time:     31.530 seconds;       Data structure size:   5.4Mb
./cuda/test_device_memset.log:CPU Time:     31.330 seconds;       Data structure size:   5.4Mb
./cuda/test_dram_device_allocated.log:CPU Time:     54.660 seconds;       Data structure size:   5.4Mb
./cuda/test_dram_host_allocated.log:CPU Time:     30.870 seconds;       Data structure size:   5.4Mb
./cuda/test_dram_load_store.log:CPU Time:     53.080 seconds;       Data structure size:   5.4Mb
./cuda/test_empty.log:CPU Time:     27.380 seconds;       Data structure size:   5.4Mb
./cuda/test_empty_parallel.log:CPU Time:     36.770 seconds;       Data structure size:   5.4Mb
./cuda/test_float_all_ops.log:CPU Time:    105.780 seconds;       Data structure size:   5.4Mb
./cuda/test_float_matrix_mul.log:CPU Time:    122.900 seconds;       Data structure size:   5.4Mb
./cuda/test_float_matrix_mul_shared_mem.log:CPU Time:    290.280 seconds;       Data structure size:   5.4Mb
./cuda/test_float_vec_add.log:CPU Time:     47.720 seconds;       Data structure size:   5.4Mb
./cuda/test_float_vec_add_shared_mem.log:CPU Time:     55.170 seconds;       Data structure size:   5.4Mb
./cuda/test_float_vec_div.log:CPU Time:     70.380 seconds;       Data structure size:   5.4Mb
./cuda/test_float_vec_exp.log:CPU Time:     85.150 seconds;       Data structure size:   5.4Mb
./cuda/test_float_vec_log.log:CPU Time:     86.000 seconds;       Data structure size:   5.4Mb
./cuda/test_float_vec_mul.log:CPU Time:     46.450 seconds;       Data structure size:   5.4Mb
./cuda/test_float_vec_sqrt.log:CPU Time:     76.020 seconds;       Data structure size:   5.4Mb
./cuda/test_host_memset.log:CPU Time:     30.220 seconds;       Data structure size:   5.4Mb
./cuda/test_log_softmax.log:CPU Time:    160.110 seconds;       Data structure size:   5.4Mb
./cuda/test_matrix_mul.log:CPU Time:    131.690 seconds;       Data structure size:   5.4Mb
./cuda/test_matrix_mul_shared_mem.log:CPU Time:    280.550 seconds;       Data structure size:   5.4Mb
./cuda/test_max_pool2d.log:CPU Time:     35.000 seconds;       Data structure size:   5.4Mb
./cuda/test_multiple_binary_load.log:CPU Time:     54.810 seconds;       Data structure size:   5.4Mb
./cuda/test_scalar_print.log:CPU Time:     30.070 seconds;       Data structure size:   5.4Mb
./cuda/test_shared_mem_load_store.log:CPU Time:     83.340 seconds;       Data structure size:   5.4Mb
./cuda/test_shared_mem.log:CPU Time:     33.160 seconds;       Data structure size:   5.4Mb
./cuda/test_softmax.log:CPU Time:    103.230 seconds;       Data structure size:   5.4Mb
./cuda/test_stack_load.log:CPU Time:     31.110 seconds;       Data structure size:   5.4Mb
./cuda/test_tile_info.log:CPU Time:     31.950 seconds;       Data structure size:   5.4Mb
./cuda/test_vec_add.log:CPU Time:     43.660 seconds;       Data structure size:   5.4Mb
./cuda/test_vec_add_parallel.log:CPU Time:     60.150 seconds;       Data structure size:   5.4Mb
./cuda/test_vec_add_parallel_multi_grid.log:CPU Time:     71.060 seconds;       Data structure size:   5.4Mb
./cuda/test_vec_add_serial_multi_grid.log:CPU Time:     68.300 seconds;       Data structure size:   5.4Mb
./cuda/test_vec_add_shared_mem.log:CPU Time:     54.590 seconds;       Data structure size:   5.4Mb
./library/test_manycore_alignment.log:CPU Time:      0.800 seconds;       Data structure size:   5.4Mb
./library/test_manycore_credits.log:CPU Time:      3.040 seconds;       Data structure size:   5.4Mb
./library/test_manycore_dmem_read_write.log:CPU Time:      0.790 seconds;       Data structure size:   5.4Mb
./library/test_manycore_dram_read_write.log:CPU Time:    120.260 seconds;       Data structure size:   5.4Mb
./library/test_manycore_eva_read_write.log:CPU Time:     21.840 seconds;       Data structure size:   5.4Mb
./library/test_manycore_init.log:CPU Time:      0.760 seconds;       Data structure size:   5.4Mb
./library/test_manycore_packets.log:CPU Time:      0.710 seconds;       Data structure size:   5.4Mb
./library/test_manycore_vcache_sequence.log:CPU Time:    123.280 seconds;       Data structure size:   5.4Mb
./library/test_printing.log:CPU Time:      0.680 seconds;       Data structure size:   5.4Mb
./library/test_read_mem_scatter_gather.log:CPU Time:      1.010 seconds;       Data structure size:   5.4Mb
./library/test_rom.log:CPU Time:      0.720 seconds;       Data structure size:   5.4Mb
./library/test_struct_size.log:CPU Time:      0.720 seconds;       Data structure size:   5.4Mb
./library/test_vcache_flush.log:CPU Time:     29.080 seconds;       Data structure size:   5.4Mb
./library/test_vcache_sequence.log:CPU Time:     14.700 seconds;       Data structure size:   5.4Mb
./library/test_vcache_simplified.log:CPU Time:      1.470 seconds;       Data structure size:   5.4Mb
./library/test_vcache_stride.log:CPU Time:      1.060 seconds;       Data structure size:   5.4Mb
./spmd/test_bsg_dram_loopback_cache.log:CPU Time:      1.890 seconds;       Data structure size:   5.4Mb
./spmd/test_bsg_loader_suite.log:CPU Time:     13.780 seconds;       Data structure size:   5.4Mb
./spmd/test_bsg_print_stat.log:CPU Time:      2.040 seconds;       Data structure size:   5.4Mb
./spmd/test_bsg_scalar_print.log:CPU Time:      1.810 seconds;       Data structure size:   5.4Mb
./spmd/test_fib.log:CPU Time:     11.140 seconds;       Data structure size:   5.4Mb
./spmd/test_putchar_stream.log:CPU Time:      2.840 seconds;       Data structure size:   5.4Mb
./spmd/test_symbol_to_eva.log:CPU Time:      0.680 seconds;       Data structure size:   5.4Mb
```
new_io_without_clock_switch.log
```
./cuda/test_barrier.log:CPU Time:     22.170 seconds;       Data structure size:   5.2Mb
./cuda/test_binary_load_buffer.log:CPU Time:     20.230 seconds;       Data structure size:   5.2Mb
./cuda/test_conv1d.log:CPU Time:     23.460 seconds;       Data structure size:   5.2Mb
./cuda/test_conv2d.log:CPU Time:     23.480 seconds;       Data structure size:   5.2Mb
./cuda/test_device_memcpy.log:CPU Time:     23.500 seconds;       Data structure size:   5.2Mb
./cuda/test_device_memset.log:CPU Time:     23.030 seconds;       Data structure size:   5.2Mb
./cuda/test_dram_device_allocated.log:CPU Time:     33.920 seconds;       Data structure size:   5.2Mb
./cuda/test_dram_host_allocated.log:CPU Time:     22.330 seconds;       Data structure size:   5.2Mb
./cuda/test_dram_load_store.log:CPU Time:     46.530 seconds;       Data structure size:   5.2Mb
./cuda/test_empty.log:CPU Time:     20.450 seconds;       Data structure size:   5.2Mb
./cuda/test_empty_parallel.log:CPU Time:     22.610 seconds;       Data structure size:   5.2Mb
./cuda/test_float_all_ops.log:CPU Time:    108.970 seconds;       Data structure size:   5.2Mb
./cuda/test_float_matrix_mul.log:CPU Time:     87.010 seconds;       Data structure size:   5.2Mb
./cuda/test_float_matrix_mul_shared_mem.log:CPU Time:    245.990 seconds;       Data structure size:   5.2Mb
./cuda/test_float_vec_add.log:CPU Time:     37.050 seconds;       Data structure size:   5.2Mb
./cuda/test_float_vec_add_shared_mem.log:CPU Time:     45.850 seconds;       Data structure size:   5.2Mb
./cuda/test_float_vec_div.log:CPU Time:     59.770 seconds;       Data structure size:   5.2Mb
./cuda/test_float_vec_exp.log:CPU Time:     74.040 seconds;       Data structure size:   5.2Mb
./cuda/test_float_vec_log.log:CPU Time:     74.380 seconds;       Data structure size:   5.2Mb
./cuda/test_float_vec_mul.log:CPU Time:     36.440 seconds;       Data structure size:   5.2Mb
./cuda/test_float_vec_sqrt.log:CPU Time:     60.880 seconds;       Data structure size:   5.2Mb
./cuda/test_host_memset.log:CPU Time:     22.560 seconds;       Data structure size:   5.2Mb
./cuda/test_log_softmax.log:CPU Time:    142.870 seconds;       Data structure size:   5.2Mb
./cuda/test_matrix_mul.log:CPU Time:     91.530 seconds;       Data structure size:   5.2Mb
./cuda/test_matrix_mul_shared_mem.log:CPU Time:    232.410 seconds;       Data structure size:   5.2Mb
./cuda/test_max_pool2d.log:CPU Time:     24.780 seconds;       Data structure size:   5.2Mb
./cuda/test_multiple_binary_load.log:CPU Time:     38.910 seconds;       Data structure size:   5.2Mb
./cuda/test_scalar_print.log:CPU Time:     22.010 seconds;       Data structure size:   5.2Mb
./cuda/test_shared_mem_load_store.log:CPU Time:     69.830 seconds;       Data structure size:   5.2Mb
./cuda/test_shared_mem.log:CPU Time:     25.390 seconds;       Data structure size:   5.2Mb
./cuda/test_softmax.log:CPU Time:     93.370 seconds;       Data structure size:   5.2Mb
./cuda/test_stack_load.log:CPU Time:     22.060 seconds;       Data structure size:   5.2Mb
./cuda/test_tile_info.log:CPU Time:     24.820 seconds;       Data structure size:   5.2Mb
./cuda/test_vec_add.log:CPU Time:     33.170 seconds;       Data structure size:   5.2Mb
./cuda/test_vec_add_parallel.log:CPU Time:     39.250 seconds;       Data structure size:   5.2Mb
./cuda/test_vec_add_parallel_multi_grid.log:CPU Time:     45.290 seconds;       Data structure size:   5.2Mb
./cuda/test_vec_add_serial_multi_grid.log:CPU Time:     45.670 seconds;       Data structure size:   5.2Mb
./cuda/test_vec_add_shared_mem.log:CPU Time:     34.320 seconds;       Data structure size:   5.2Mb
./library/test_manycore_alignment.log:CPU Time:      0.950 seconds;       Data structure size:   5.2Mb
./library/test_manycore_credits.log:CPU Time:      2.920 seconds;       Data structure size:   5.2Mb
./library/test_manycore_dmem_read_write.log:CPU Time:      0.790 seconds;       Data structure size:   5.2Mb
./library/test_manycore_dram_read_write.log:CPU Time:     14.700 seconds;       Data structure size:   5.2Mb
./library/test_manycore_eva_read_write.log:CPU Time:     24.490 seconds;       Data structure size:   5.2Mb
./library/test_manycore_init.log:CPU Time:      0.800 seconds;       Data structure size:   5.2Mb
./library/test_manycore_packets.log:CPU Time:      0.830 seconds;       Data structure size:   5.2Mb
./library/test_manycore_vcache_sequence.log:CPU Time:     14.980 seconds;       Data structure size:   5.2Mb
./library/test_printing.log:CPU Time:      0.770 seconds;       Data structure size:   5.2Mb
./library/test_read_mem_scatter_gather.log:CPU Time:      0.900 seconds;       Data structure size:   5.2Mb
./library/test_rom.log:CPU Time:      0.810 seconds;       Data structure size:   5.2Mb
./library/test_struct_size.log:CPU Time:      0.780 seconds;       Data structure size:   5.2Mb
./library/test_vcache_flush.log:CPU Time:     25.390 seconds;       Data structure size:   5.2Mb
./library/test_vcache_sequence.log:CPU Time:     13.460 seconds;       Data structure size:   5.2Mb
./library/test_vcache_simplified.log:CPU Time:      0.850 seconds;       Data structure size:   5.2Mb
./library/test_vcache_stride.log:CPU Time:      1.100 seconds;       Data structure size:   5.2Mb
./spmd/test_bsg_dram_loopback_cache.log:CPU Time:      1.250 seconds;       Data structure size:   5.2Mb
./spmd/test_bsg_loader_suite.log:CPU Time:      7.520 seconds;       Data structure size:   5.2Mb
./spmd/test_bsg_print_stat.log:CPU Time:      1.240 seconds;       Data structure size:   5.2Mb
./spmd/test_bsg_scalar_print.log:CPU Time:      1.330 seconds;       Data structure size:   5.2Mb
./spmd/test_fib.log:CPU Time:      7.500 seconds;       Data structure size:   5.2Mb
./spmd/test_putchar_stream.log:CPU Time:      2.220 seconds;       Data structure size:   5.2Mb
./spmd/test_symbol_to_eva.log:CPU Time:      0.730 seconds;       Data structure size:   5.2Mb
```

Xilinx IP IO with clock switching:
```
./cuda/test_barrier.log:CPU Time:     80.420 seconds;       Data structure size:   5.7Mb
./cuda/test_binary_load_buffer.log:CPU Time:     74.640 seconds;       Data structure size:   5.7Mb
./cuda/test_conv1d.log:CPU Time:    117.320 seconds;       Data structure size:   5.7Mb
./cuda/test_conv2d.log:CPU Time:     86.920 seconds;       Data structure size:   5.7Mb
./cuda/test_device_memcpy.log:CPU Time:     80.890 seconds;       Data structure size:   5.7Mb
./cuda/test_device_memset.log:CPU Time:     81.010 seconds;       Data structure size:   5.7Mb
./cuda/test_dram_device_allocated.log:CPU Time:    101.110 seconds;       Data structure size:   5.7Mb
./cuda/test_dram_host_allocated.log:CPU Time:     80.920 seconds;       Data structure size:   5.7Mb
./cuda/test_dram_load_store.log:CPU Time:    123.490 seconds;       Data structure size:   5.7Mb
./cuda/test_empty.log:CPU Time:     74.790 seconds;       Data structure size:   5.7Mb
./cuda/test_empty_parallel.log:CPU Time:     77.330 seconds;       Data structure size:   5.7Mb
./cuda/test_float_all_ops.log:CPU Time:    208.120 seconds;       Data structure size:   5.7Mb
./cuda/test_float_matrix_mul.log:CPU Time:    158.900 seconds;       Data structure size:   5.7Mb
./cuda/test_float_matrix_mul_shared_mem.log:CPU Time:    328.920 seconds;       Data structure size:   5.7Mb
./cuda/test_float_vec_add.log:CPU Time:    107.290 seconds;       Data structure size:   5.7Mb
./cuda/test_float_vec_add_shared_mem.log:CPU Time:    117.190 seconds;       Data structure size:   5.7Mb
./cuda/test_float_vec_div.log:CPU Time:    136.600 seconds;       Data structure size:   5.7Mb
./cuda/test_float_vec_exp.log:CPU Time:    163.890 seconds;       Data structure size:   5.7Mb
./cuda/test_float_vec_log.log:CPU Time:    162.460 seconds;       Data structure size:   5.7Mb
./cuda/test_float_vec_mul.log:CPU Time:    106.640 seconds;       Data structure size:   5.7Mb
./cuda/test_float_vec_sqrt.log:CPU Time:    155.700 seconds;       Data structure size:   5.7Mb
./cuda/test_host_memset.log:CPU Time:     80.880 seconds;       Data structure size:   5.7Mb
./cuda/test_log_softmax.log:CPU Time:    244.850 seconds;       Data structure size:   5.7Mb
./cuda/test_matrix_mul.log:CPU Time:    168.800 seconds;       Data structure size:   5.7Mb
./cuda/test_matrix_mul_shared_mem.log:CPU Time:    320.010 seconds;       Data structure size:   5.7Mb
./cuda/test_max_pool2d.log:CPU Time:     85.640 seconds;       Data structure size:   5.7Mb
./cuda/test_multiple_binary_load.log:CPU Time:    146.550 seconds;       Data structure size:   5.7Mb
./cuda/test_scalar_print.log:CPU Time:     81.750 seconds;       Data structure size:   5.7Mb
./cuda/test_shared_mem_load_store.log:CPU Time:    151.740 seconds;       Data structure size:   5.7Mb
./cuda/test_shared_mem.log:CPU Time:     84.840 seconds;       Data structure size:   5.7Mb
./cuda/test_softmax.log:CPU Time:    188.030 seconds;       Data structure size:   5.7Mb
./cuda/test_stack_load.log:CPU Time:     81.000 seconds;       Data structure size:   5.7Mb
./cuda/test_tile_info.log:CPU Time:     91.010 seconds;       Data structure size:   5.7Mb
./cuda/test_vec_add.log:CPU Time:    102.760 seconds;       Data structure size:   5.7Mb
./cuda/test_vec_add_parallel.log:CPU Time:    108.290 seconds;       Data structure size:   5.7Mb
./cuda/test_vec_add_parallel_multi_grid.log:CPU Time:    119.400 seconds;       Data structure size:   5.7Mb
./cuda/test_vec_add_serial_multi_grid.log:CPU Time:    118.030 seconds;       Data structure size:   5.7Mb
./cuda/test_vec_add_shared_mem.log:CPU Time:     96.340 seconds;       Data structure size:   5.7Mb
./library/test_manycore_alignment.log:CPU Time:      1.020 seconds;       Data structure size:   5.7Mb
./library/test_manycore_credits.log:CPU Time:     12.010 seconds;       Data structure size:   5.7Mb
./library/test_manycore_dmem_read_write.log:CPU Time:      1.200 seconds;       Data structure size:   5.7Mb
./library/test_manycore_dram_read_write.log:CPU Time:    132.930 seconds;       Data structure size:   5.7Mb
./library/test_manycore_eva_read_write.log:CPU Time:    103.620 seconds;       Data structure size:   5.7Mb
./library/test_manycore_init.log:CPU Time:      1.270 seconds;       Data structure size:   5.7Mb
./library/test_manycore_packets.log:CPU Time:      1.160 seconds;       Data structure size:   5.7Mb
./library/test_manycore_vcache_sequence.log:CPU Time:    136.630 seconds;       Data structure size:   5.7Mb
./library/test_printing.log:CPU Time:      1.130 seconds;       Data structure size:   5.7Mb
./library/test_read_mem_scatter_gather.log:CPU Time:      1.190 seconds;       Data structure size:   5.7Mb
./library/test_rom.log:CPU Time:      1.180 seconds;       Data structure size:   5.7Mb
./library/test_struct_size.log:CPU Time:      0.960 seconds;       Data structure size:   5.7Mb
./library/test_vcache_flush.log:CPU Time:     96.770 seconds;       Data structure size:   5.7Mb
./library/test_vcache_sequence.log:CPU Time:     55.730 seconds;       Data structure size:   5.7Mb
./library/test_vcache_simplified.log:CPU Time:      1.510 seconds;       Data structure size:   5.7Mb
./library/test_vcache_stride.log:CPU Time:      2.170 seconds;       Data structure size:   5.7Mb
./python/test_python.log:CPU Time:      1.040 seconds;       Data structure size:   5.7Mb
./spmd/test_bsg_dram_loopback_cache.log:CPU Time:      3.070 seconds;       Data structure size:   5.7Mb
./spmd/test_bsg_loader_suite.log:CPU Time:     29.120 seconds;       Data structure size:   5.7Mb
./spmd/test_bsg_print_stat.log:CPU Time:      2.780 seconds;       Data structure size:   5.7Mb
./spmd/test_bsg_scalar_print.log:CPU Time:      2.520 seconds;       Data structure size:   5.7Mb
./spmd/test_fib.log:CPU Time:     17.360 seconds;       Data structure size:   5.7Mb
./spmd/test_putchar_stream.log:CPU Time:      6.940 seconds;       Data structure size:   5.7Mb
./spmd/test_symbol_to_eva.log:CPU Time:      1.000 seconds;       Data structure size:   5.7Mb

```

**Comment by drichmond** [comment]
Are you sure that the Xilinx data is with the clock switching?

I only ask because it's a lot slower than I expected

**Comment by drichmond** [comment]
![image](https://user-images.githubusercontent.com/6378612/75394461-0716bc00-58a5-11ea-8f7f-75772f1351d8.png)

**Comment by drichmond** [comment]
This data suggests that we should remove clock switching and use Leonard's new IO because it's faster than the old IO + clock switching. In fact, it's nearly as fast as my current solution.

**Comment by leonardxiang** [comment]
> This data suggests that we should remove clock switching and use Leonard's new IO because it's faster than the old IO + clock switching. In fact, it's faster than PR 509 even though it generates vpds (@leonardxiang your regression run generated valid vpd files, right?)

Yes, I used make regression. And ran it on master of bladerunner.
Nice table BTW!

**Comment by drichmond** [comment]
![image](https://user-images.githubusercontent.com/6378612/75396019-63c7a600-58a8-11ea-8fc1-3dfb1e679800.png)

I didn't believe the data until I ran it myself... but it looks like the combination of my PR and Leonard's is producing astronomical speed-ups. We're getting a 5x speedup over our old regression flow (apples to oranges) and a 2.8x speed-up in my PR (apples to apples)

**Comment by drichmond** [comment]
@proftaylor, @mrutt92 your thoughts? Should we axe clock switching?

I'd be curious to see the dramsim3 back-door timings. I believe that will close the gap between clock-switching and non-clock-switching runs.

For example, If we take test_empty (~1k instructions, no data, just loads an empty program and launches a kernel) as the baseline for loading the icache and doing nothing else, the icache initialization overhead is only 1s for a 4x4 tile group. I'd expect that for every run after adding dramsim3.

That's not much overhead. Or, another way, I'm willing to pay that overhead on F1 emulation if we can remove the Xilinx clock-switching IP.

**Comment by drichmond** [comment]
This might be the next issue, but food for thought: Why do all of the "simple kernels" with minimal computation take about 10-15 seconds? What part of start/up and initialization takes that long? 

Is it icache initialization? (I don't think so, see above)
Is it vcache initialization?

Diagnosing (and fixing?) that would make a lot of development lives easier.

**Comment by taylor-bsg** [comment]
It would be nice to quantify the relative performance in terms of Packets per wall clock time using a simple I/O micro benchmark? I think we generally are trying to quantify large write bursts, right? This would also help us identify other bottlenecks.


M

**Comment by taylor-bsg** [comment]
That will help us understand the overhead based on binary properties — some kind of fixed overhead plus some binary-dependent overheads.

**Comment by drichmond** [comment]
Yeah - that's a good idea. We can measure this already, but we don't have a test explicitly for it. 

In said test, It'd be good to measure:

- Time to load i$ for all tiles
- Time to load each v$ and a bank (?) work of DRAM
- Time to launch an empty tile group

Anything else interesting that I'm missing?

**Comment by taylor-bsg** [comment]
Since number of tiles and numbers of vcaches vary, it would be nice to have
a raw packet rate, and the. we can analytically compute the amount of time
by computing how much state needs to be loaded for a particular
configuration and binary, including the statedul components you listed below

**Comment by drichmond** [comment]
Filed an issue

**Comment by drichmond** [comment]
But even the "raw packet rate" is subject to the architecture. Perhaps not with writes, but with reads, yes. How far is your packet expected to traverse? Is it going to memory, i$, or Dmem?

**Comment by drichmond** [comment]
I'm not disagreeing with you, I'm just enumerating the complexity

**Comment by taylor-bsg** [comment]
If the system is reasonably optimized, these should all be pretty close for
writes. For reads, if we support some concurrency in terms of having
multiple outstanding reads, which should then also be pretty close?

**Comment by drichmond** [comment]
I'm going to close this. With new verilator simulation removing the clock-crossing in F1 is somewhat superfluous (and it does have some useful benefits)

**Comment by proftaylor** [comment]
What about VCS though — presumably with the Verilator improvements we can
get faster I/O in VCS?

**Comment by drichmond** [comment]
Absolutely possible! Integrating my new DPI interfaces isn't possible in the F1 top-level and the clock crossing is necessary for performant cosimulation of the F1 design. Therefore, I closed this since it refers to the top-level F1 file.

If we want a new top-level for VCS it can be created, but it will have to be in a different top level than the one we use for F1 (possibly the same file as the *new* one we use for Verilator). 

There are a few other issues that need to be addressed before we can simulate with the new DPI interfaces in verilator. I'd be happy to describe them in our next meeting.
